tests/test_admin_views.py

```diff
@@ -0,0 +1,156 @@
+        response = self.c.get(reverse('admin_main_add'))
+        assert response.status_code == 200
+        # Assert the subject will be added to the top level
+        assert response.context['browse_len'] == 0
+        assert response.context['saved'] is None
+        assert response.templates[0].name == 'subjects/add_subject.html'
+
+    def test_subject_add_to_sub_id(self):
+        """Test the AddSubject form for new subject will add it as a sub-category
+         to the current subject."""
+        subject = factories.SubjectFactory.create()
+        response = self.c.get(reverse('admin_add', kwargs={'sub_id': subject.id}))
+        # Assert the new subject will be added as a sub category of 'subject'.
+        assert response.context['browse_len'] == 1
+        assert response.context['browse_string'] == subject.name
+        assert response.context['saved'] is None
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/add_subject.html'
+
+    def test_subject_add_post_with_sub_id(self):
+        """Test the new subject is added."""
+        subject = factories.SubjectFactory.create()
+        name = 'foo'
+        with mock.patch('subjects.admin_views.AddSubject') as mock_form:
+            mock_form.return_value.is_valid.return_value = True
+            mock_form.return_value.cleaned_data = {'name': name,
+                                                   'keywords': 'test keywords',
+                                                   'notes': 'test notes'}
+            response = self.c.post(reverse('admin_add', kwargs={'sub_id': subject.id}))
+
+        assert response.context['saved'].name == name
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/add_subject.html'
+
+
+@pytest.mark.django_db
+class TestSubjectModify(TestCase):
+
+    def setUp(self):
+        self.admin_user = User.objects.create_superuser(
+            'test_user', 'test_user@test.com', 'password')
+        self.admin_user.save()
+        self.c = Client()
+        self.c.force_login(self.admin_user)
+
+    def test_subject_modify(self):
+        """Test main modify menu is displayed."""
+        response = self.c.get(reverse('admin_main_modify'))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/modify_subject.html'
+        assert response.context['modified'] is None
+
+    def test_subject_modify_post_with_sub_id(self):
+        """Test the subject is modified."""
+        subject = factories.SubjectFactory.create()
+        with mock.patch('subjects.admin_views.ModifySubject') as mock_form:
+            mock_form.return_value.is_valid.return_value = True
+            mock_form.return_value.cleaned_data = {'keywords': 'modified_keyword',
+                                                   'notes': 'modified_notes'}
+            response = self.c.post(reverse('admin_modify', kwargs={'sub_id': subject.id}))
```

Comment:
Similarly, I think we could do without the mock of `ModifySubject`:
```suggestion
        response = self.c.post(reverse('admin_modify', kwargs={'sub_id': subject.id}),
                               data={'keywords': 'modified_keyword', 'notes': 'modified_notes'})
```

Reply:
Removed mocking Subject forms as you suggested.